sdk/messaging/azservicebus/receiver.go

```diff
@@ -507,15 +507,47 @@ func (e *entity) SetSubQueue(subQueue SubQueue) error {
 	return fmt.Errorf("unknown SubQueue %d", subQueue)
 }
 
-func createReceiverLink(ctx context.Context, session internal.AMQPSession, linkOptions []amqp.LinkOption) (internal.AMQPReceiverCloser, error) {
-	amqpReceiver, err := session.NewReceiver(linkOptions...)
-
-	if err != nil {
-		tab.For(ctx).Error(err)
-		return nil, err
+func createReceiverLink(ctx context.Context, session amqpwrap.AMQPSession, linkOptions []amqp.LinkOption) (internal.AMQPReceiverCloser, error) {
+	// If you're doing an AcceptNextSession it's possible for this call to take a long time before timing out
+	// on its own (it's by design - it's waiting for any empty session to become available).
+	type ret = struct {
+		Receiver internal.AMQPReceiverCloser
+		Err      error
 	}
 
-	return amqpReceiver, nil
+	done := make(chan ret, 1)
```

Comment:
I don't think this needs to be a buffered channel does it?

Reply:
You were correct! In fact, it was better unbuffered.